### ISSUE TITLE
office 365 Links Grant File Access errors

### ISSUE THREAD

**Issue by msj107** [jmcnamara/excel-writer-xlsx#248]
Since being forced to ugrade to Office 360 when I create links I get a pop up error box for Grant File Access. Having lots of links this pop up is very annoying. I had no issues before with my Excel 2008.

Anyone else see this?

**Comment by jmcnamara** [comment]
I haven’t seen that issue or received any similar reports. Could you add a small minimal program that demonstrates the issue.

**Comment by msj107** [comment]
my program is pretty big let me see about making something simple 
I have had microsoft support doing one on one help trying to figure it out and they cant
Thanks!

**Comment by jmcnamara** [comment]
Start simple. Does it occur with a program that produces just one hyperlink. Is the hyperlink internal, external or to the web, etc.

**Comment by msj107** [comment]
using write_url and an external to a file found locally on the computer drive

$worksheet_summary->write_url( "$char$temp", "external:$PWD/$year/$mon/$cat.xlsx", $url_format, "\$$$YEAR_MONTH_CAT_TOTAL", $format_merged_right );

**Comment by jmcnamara** [comment]
So if you turned that into a working example like the following does it raise the warning?

```perl
#!/usr/bin/perl

use strict;
use warnings;
use Excel::Writer::XLSX;

print 'Version = ', Excel::Writer::XLSX->VERSION, "\n";

# Create one workbook.
my $workbook = Excel::Writer::XLSX->new( 'cat.xlsx' );
my $worksheet = $workbook->add_worksheet();

$workbook->close();

# Create a second workbook that links to the first.
$workbook = Excel::Writer::XLSX->new( 'dog.xlsx' );
$worksheet = $workbook->add_worksheet();

$worksheet->write( 0, 0, 'external:cat.xlsx' );

$workbook->close();

```

**Comment by jmcnamara** [comment]
Ok. I was able to reproduce the issue with Excel 16.33 for the Mac but not with earlier versions.

Are you on macOS as well?

If so it look like (from a Google search) that this is a issue introduced by macOS sandboxing.

There are some suggested workaround like this one: https://support.office.com/en-us/article/Excel-prompts-you-to-grant-access-to-files-and-then-hangs-or-shows-a-warning-that-it-cannot-open-the-xlsx-file-because-the-format-or-extension-is-not-valid-cc9e40d2-e166-4021-8b61-2294e15e9e18

Either way this isn't an Excel::Writer::XLSX issue since it creates hyperlinks exactly like Excel does.

So I'm going to close this.

**Comment by msj107** [comment]
I thanks you for taking the time to look into this, yes it is on Mac and I did find this article previously and unfortunately it does not make the access grant issue go away. I do agree it did not appear to be a perl coding issue as when I am in excel and look at the link it looks correct.

THanks
Michael




On January 21, 2020 at 3:30 PM, John McNamara <notifications@github.com> wrote:

Ok. I was able to reproduce the issue with Excel 16.33 for the Mac but not with earlier versions.
Are you on macOS as well?
If so it look like (from a Google search) that this is a issue introduced by macOS sandboxing.
There are some suggested workaround like this one: https://support.office.com/en-us/article/Excel-prompts-you-to-grant-access-to-files-and-then-hangs-or-shows-a-warning-that-it-cannot-open-the-xlsx-file-because-the-format-or-extension-is-not-valid-cc9e40d2-e166-4021-8b61-2294e15e9e18
Either way this isn't an Excel::Writer::XLSX issue since it creates hyperlinks exactly like Excel does.
So I'm going to close this.
